pkg/network/ebpf/c/protocols/tls/tls.h

```diff
@@ -1,132 +1,459 @@
-    if (hdr->length < sizeof(tls_hello_message_t)) {
+
+    // Read Cipher Suites Length (2 bytes)
+    if (offset + CIPHER_SUITES_LENGTH > data_end) {
         return false;
     }
+    __u16 cipher_suites_length;
+    if (bpf_skb_load_bytes(skb, offset, &cipher_suites_length, CIPHER_SUITES_LENGTH) < 0) {
+        return false;
+    }
+    cipher_suites_length = bpf_ntohs(cipher_suites_length);
+    offset += CIPHER_SUITES_LENGTH;
+
+    // Skip Cipher Suites
+    offset += cipher_suites_length;
 
-    // Getting the tls hello message header.
-    tls_hello_message_t msg = *(tls_hello_message_t *)(buf + sizeof(tls_record_header_t));
-    // If the message is not a CLIENT_HELLO or SERVER_HELLO, we don't attempt to classify.
-    if (msg.handshake_type != TLS_HANDSHAKE_CLIENT_HELLO && msg.handshake_type != TLS_HANDSHAKE_SERVER_HELLO) {
+    // Read Compression Methods Length (1 byte)
+    if (offset + COMPRESSION_METHODS_LENGTH > data_end) {
+        return false;
+    }
+    __u8 compression_methods_length;
+    if (bpf_skb_load_bytes(skb, offset, &compression_methods_length, COMPRESSION_METHODS_LENGTH) < 0) {
         return false;
     }
-    // Converting the fields to host byte order.
-    __u32 length = msg.length[0] << 16 | msg.length[1] << 8 | msg.length[2];
-    // TLS handshake message length should be equal to the record header length minus the size of the hello message
-    // header.
-    if (length + TLS_HELLO_MESSAGE_HEADER_SIZE != hdr->length) {
+    offset += COMPRESSION_METHODS_LENGTH;
+
+    // Skip Compression Methods
+    offset += compression_methods_length;
+
+    // Check if extensions are present
+    if (offset + EXTENSION_LENGTH_FIELD > data_end) {
+        return false;
+    }
+
+    // Read Extensions Length (2 bytes)
+    __u16 extensions_length;
+    if (bpf_skb_load_bytes(skb, offset, &extensions_length, EXTENSION_LENGTH_FIELD) < 0) {
+        return false;
+    }
+    extensions_length = bpf_ntohs(extensions_length);
+    offset += EXTENSION_LENGTH_FIELD;
+
+    if (offset + extensions_length > data_end) {
         return false;
     }
 
-    msg.version = bpf_ntohs(msg.version);
-    return is_valid_tls_version(msg.version) && msg.version >= hdr->version;
+    __u64 extensions_end = offset + extensions_length;
+
+    return parse_tls_extensions(skb, &offset, data_end, extensions_end, tags, true);
 }
 
-// is_tls checks if the given buffer is a valid TLS record header. We are
-// currently checking for two types of record headers:
-// - TLS Handshake record headers
-// - TLS Application Data record headers
-static __always_inline bool is_tls(const char *buf, __u32 buf_size, __u32 skb_len) {
-    if (buf_size < sizeof(tls_record_header_t)) {
+// parse_server_hello parses the ServerHello message and populates tags
+static __always_inline bool parse_server_hello(struct __sk_buff *skb, __u64 offset, __u32 data_end, tls_info_t *tags) {
```

Comment:
please add a link to the RFC structs message so people can refer to the decoding logic here